### DISCUSSION TITLE
Possible to collect counters/gauges/histograms from SQL operations?

### DISCUSSION THREAD

**Discussion by ToddBeaulieu** [open-telemetry/opentelemetry-dotnet#4408] [Q&A] [answered]
Very new to OTEL. Working on custom metrics for services, ultimately correlating to logs and traces. For drilling in, it's obvious how incredible trace data is, but I would have assumed that we'd also have access to simple metrics for operations so that we wouldn't need to instrument database calls to collect counters and durations. I'd use such simple metrics for light-weight, high-level monitoring.

Is that possible? Or, would I need to somehow generate such metrics FROM the trace data? Or (again), am I thinking about this wrong, and it's easy to simply extract "metrics" from the trace data for dashes?

Thanks!

**Comment by cijothomas** [comment, marked as answer]
Otel semantic conventions for metrics from database is very limited. IIRC, it only has a metric about the number of open connections. There are no metrics defined for the duration of the sql calls or the count of calls etc.

**Reply by ToddBeaulieu** [reply]
Thank you. Do you recommend instrumenting .net code, wrapping calls with historgrams, or is it easy (and performant) to grab those stats for a dashboard? i.e. a histogram panel for all or key db calls from a service.
